Assets/MRTK/SDK/Features/UX/Scripts/Pointers/BaseControllerPointer.cs

```diff
@@ -227,7 +230,6 @@ public override IMixedRealityController Controller
 
                 if (base.Controller != null && this != null)
                 {
-                    PointerName = $"{Handedness}_{gameObject.name}";
```

Comment:
I think moving this out of the `Controller` setter might cause unexpected pointer names in the case where they're recycled and reused.

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/965f566906dc15b57b9fa8f2dd6c713065685670/Assets/MRTK/Core/Providers/BaseInputDeviceManager.cs#L200-L220

in that case, we'll reuse a pointer GameObject which will retain its previous name, which might have been for a differently-handed controller

Reply:
Currently, with the name update in the setter, the name will update when a reused pointer is updated at, for example: https://github.com/microsoft/MixedRealityToolkit-Unity/blob/29480068f483ea42422949e253fcf46019770441/Assets/MRTK/Providers/XRSDK/XRSDKDeviceManager.cs#L205-L208

Reply:
Unfortunately this led to the repeating handedness bug we observed earlier. I suppose the best way to go about addressing this is to have another variable that denotes the pointerTypeName? Though, I'm not super enthuthiastic about setting that initially and then never touching it again. I also wasn't super thrilled that the naming methodology had started to get fragmented across the many different pointers.

Reply:
Yeah, that's definitely fair! I wonder if we could split on `_` and replace the first value with our new handedness or something?

Reply:
>I also wasn't super thrilled that the naming methodology had started to get fragmented across the many different pointers.

Also this though, yeah :(

Reply:
> Yeah, that's definitely fair! I wonder if we could split on `_` and replace the first value with our new handedness or something?

I worry that we start relying too much on the naming structure of the object that's getting passed in, and the extra variable approach is slightly more appealing to me in this case...

Reply:
Followup PR here https://github.com/microsoft/MixedRealityToolkit-Unity/pull/9371